ODIN_II/SRC/netlist_create_from_ast.cpp

```diff
@@ -1034,6 +1034,42 @@ nnet_t* define_nets_with_driver(ast_node_t* var_declare, char* instance_name_pre
 
             if (var_declare->types.variable.initial_value) {
                 new_net->initial_value = (init_value_e)var_declare->types.variable.initial_value->get_bit_from_lsb(i);
+
+                npin_t* new_pad_pin = allocate_npin();
+            
+                allocate_more_output_pins(verilog_netlist->pad_node, 1);
+                add_output_pin_to_node(verilog_netlist->pad_node, new_pad_pin, verilog_netlist->pad_node->num_output_pins - 1);
+                add_driver_pin_to_net(new_net, new_pad_pin);
+            } else {
```

Comment:
Ok, lets start again.
Nets always have an an initial_value, and the default is 3. (also known as unknown see the blif specification)
the else should simply be
```
new_net->initial_value = undefined;
```

once you figured the initial_value. you don't touch it again. the bits of code bellow is wrong here. it need to be moved later during netlist cleanup. that is what you have to do.

why?

because the init value does not drive the circuit ever. 
the bits of code you have here should be created during netlist cleanup, if the cleanup is about to destroy a latch with a defined initial_value, just connect it to unconn and stop sweeping.

that ^ is the solution.

Reply:
now (this is a separate issue, but needs to be addressed here so you can pass the tests) the primary output pins are dropped in output_blif I believe. make sure we print them.